File: 198. House Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int prevMax = 0, curMax = 0;
        for (int i = 0; i < nums.size(); i++) {
            int tmp = curMax;
            curMax = max(prevMax + nums[i], curMax);
            prevMax = tmp;
        }
        return curMax;
    }
};

File: java/198. House Robber.java
// O(n)
class Solution {
    public int rob(int[] nums) {
        int p2 = 0, p1 = 0;
        for (int i = 0; i < nums.length; ++i) {
            int temp = p1;
            p1 = Math.max(p2 + nums[i], p1);
            p2 = temp;
        }
        return p1;
    }
}
